vignettes/examples.Rmd

```diff
@@ -327,6 +332,79 @@ ggplot(dat) +
   theme(legend.position = "none")
 ```
 
+Set `segment.square`to `FALSE` to get oblique curves, and `segment.inflect` to `TRUE` to introduce an inflection point.
+
+```{r line_inflect_and_square, echo=TRUE, fig.width=6, fig.height=4}
+set.seed(42)
+
+cars_subset <- head(mtcars, 5)
+cars_subset$car <- rownames(cars_subset)
+
+cars_subset_curves <- cars_subset[rep(seq_len(nrow(cars_subset)), times = 4), ]
+cars_subset_curves$square <- rep(c(TRUE, FALSE), each = nrow(cars_subset) * 2)
+cars_subset_curves$inflect <- rep(c(TRUE, FALSE, TRUE, FALSE), each = nrow(cars_subset))
+
+ggplot(cars_subset_curves, aes(y = wt, x = 1, label = car)) +
+  facet_grid(square ~ inflect, labeller = labeller(.default = label_both)) +
+  geom_point(color = "red") +
+  ylim(1, 4.5) +
+  xlim(1, 1.375) +
+  geom_text_repel(
+    aes(
+      segment.square  = square,
+      segment.inflect = inflect,
+    ),
+    force             = 0.5,
+    nudge_x           = 0.15,
+    direction         = "y",
+    hjust             = 0,
+    segment.size      = 0.2,
+    segment.curvature = -0.1
+  ) +
+  theme(
+    axis.line.x  = element_blank(),
+    axis.ticks.x = element_blank(),
+    axis.text.x  = element_blank(),
+    axis.title.x = element_blank()
+  )
+```
+
+Use `segment.shape` to adjust the interpolation of the control points:
+
+```{r line_shape, echo=TRUE, fig.width=5, fig.height=10}
+set.seed(42)
+
+cars_subset_shapes <- cars_subset[rep(seq_len(nrow(cars_subset)), times = 5), ]
+cars_subset_shapes$shape <- rep(c(-1, -0.5, 0, 0.5, 1), each = nrow(cars_subset))
+
+ggplot(cars_subset_curves, aes(y = wt, x = 1, label = car)) +
+  facet_wrap('shape', labeller = labeller(.default = label_both), ncol = 1) +
+  geom_point(color = "red") +
+  ylim(1, 4.5) +
+  xlim(1, 1.375) +
+  geom_text_repel(
+    aes(
+      segment.shape   = shape
+    ),
+    force             = 0.5,
+    nudge_x           = 0.25,
+    direction         = "y",
+    hjust             = 0,
+    segment.size      = 0.2,
+    segment.curvature = -0.6,
+    segment.angle = 45,
+    segment.ncp = 2,
+    segment.square  = FALSE,
+    segment.inflect = TRUE
+  ) +
+  theme(
+    axis.line.x  = element_blank(),
+    axis.ticks.x = element_blank(),
+    axis.text.x  = element_blank(),
+    axis.title.x = element_blank()
+  )
```

Comment:
This is now used in four examples; for brevity it could be replaced by something like:

```R
clean_x_axis_theme <- theme(
    axis.line.x  = element_blank(),
    axis.ticks.x = element_blank(),
    axis.text.x  = element_blank(),
    axis.title.x = element_blank()
  )
```

and then `clean_x_axis_theme` would be added to any example which needs it